### ISSUE TITLE
[BUG] Multi-User Generation Leakage

### ISSUE THREAD

**Issue by alicat22** [theroyallab/tabbyAPI#107]
**Describe the bug**
Person 1 generates. Person 2 generates immediately after. Person 2 gets Person 1's generation or part of Person 1's generation.

**To Reproduce**
Steps to reproduce the behaviour:
1. Generate from frontend 1
2. Generate from frontend 2 via a tunnel
4. See frontend 1's response on frontend 2

*Note, this may require multiple attempts and spamming. It should be relatively easy to reproduce.

**Expected behavior**
Person 1 should receive the response and should not affect Person 2's responses in any way.

**Logs**
```
INFO:     <|start_header_id|>user<|end_header_id|>
INFO:
INFO:     Alexandria: genning something<|eot_id|><|start_header_id|>assistant<|end_header_id|>
INFO:
INFO:     Gensokyo:
INFO:     Response: 
INFO:      "Excellent. Follow @alice;  I'm, and follow this alice and, the story.
INFO:     Metrics: 22 tokens generated in 1.45 seconds (15.15 T/s, context 2132 tokens)
INFO:     10.32.2.251:0 - "POST /v1/completions HTTP/1.1" 200
INFO:     Response: 
INFO:      this is my name you worthy to join my adventures. Or, not, is.user:
INFO:
INFO:     Gensokyo: Alice Margatroid: That's quite cryptic, but alright. If you follow me, I promise to show you the most thrilling adventures!user:
INFO:
INFO:     Gensokyo: Alexandria: Cool, let's do it!
INFO:     Metrics: 75 tokens generated in 2.41 seconds (31.17 T/s, context 1019 tokens)
INFO:     127.0.0.1:49939 - "POST /v1/completions HTTP/1.1" 200
INFO:     10.32.1.187:0 - "POST /v1/completions HTTP/1.1" 200
```
Context is from Person 1. The first part of the response was sent to Person 2:
```
INFO:     Response: 
INFO:      "Excellent. Follow @alice;  I'm, and follow this alice and, the story.
```
The second part of the response was sent to Person 1:
```
INFO:     Response: 
INFO:      this is my name you worthy to join my adventures. Or, not, is.user:
```

**System info** (Bugs without this information will go lower on our priority list!)
 - OS: Win 11
 - Python version: Python 3.11.9
 - CUDA/ROCm version:
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0

**Comment by mku-wedoai** [comment]
I would kindly ask to prioritise this bug as it is holding up my work on an important project.

**Comment by bdashore3** [comment]
This should be cleaned up and working in c474076b22edce22dae150917e0711f8cf63ec39. The bug is not easily reproducible and only occurred with non-streaming + streaming prompts.

I'm keeping this issue open since Exl2 has a different generator in the works and should close any complaints.

**Comment by bdashore3** [comment]
Resolved in #109
